### ISSUE TITLE
The og:image is not recognized by facebook nor twitter

### ISSUE THREAD

**Issue by Laurent3170** [mprins/dokuwiki-plugin-socialcards#14]
When I set the image, full url or just name, the meta is filled by "https://wikidurable.org/lib/exe/fetch.php/logo.png", which works but is not recognized by facebook and twitter when I fetch the scrape information.

**Comment by Laurent3170** [comment]
I have set the redirections for DW to be done via .htaccess, so even with the url https://wikidurable.org/_media/wiki/logo.png, the file can't be accessed by fb...

**Comment by mprins** [comment]
I'm not sure what you mean with
> when I fetch the scrape information

both Twitter and Facebook provide tooling to activate and check metadata in documents, eg. https://dev.twitter.com/docs/cards/validation/validator (afaik Twitter does not use OpenGraph) and http://developers.facebook.com/tools/debug/

the `og:image` tag is a required element, see: http://ogp.me/#metadata if it's not working for you you'll need to talk to them I think. There may be an issue using TLS (https) on the image url; did you try without?

BTW I seems you did not update your template? https://www.dokuwiki.org/plugin:socialcards#configuration_and_settings

**Comment by Laurent3170** [comment]
Hi,
thanks for your answer.
Yes I have used twitter and facebook tools, they don't find the image. facebook said 

> og:image could not be downloaded or is too small
I have tried with an image from another site which I now is ok and recognized, but same result. Just tried with http://... but again same result. So I will look at 

> BTW I seems you did not update your template? https://www.dokuwiki.org/plugin:socialcards#configuration_and_settings

**Comment by Laurent3170** [comment]
So I think it's a problem with TLS as you mentionned because the http is 301 redirected to https...
I haven't found where to modify the meta in the templates, if you can let me know where it is, it would be great :smile:

**Comment by mprins** [comment]
> I haven't found where to modify the meta in the templates, if you can let me know where it is, it would be great :smile:

You just need to edit a few files (main.php, detail.php and mediamanager.php) in`.../dokuwiki/lib/tpl/<your template>` to add those namespace prefixes


as SSL is concerned, it seems there is a way to specify a secure url `og:image:secure_url` see: http://ogp.me/#structured

**Comment by Laurent3170** [comment]
Thanks !
I'll have a look at it.

**Comment by Laurent3170** [comment]
So...
I went into your code, because of the need to add the secure_url.
Doing this I saw that your fallback image... is a fallback image :sweat_smile: where I was thinking using it as THE default image, the one I want to see for the whole site.
Regarding this, I have updated the french translation to add the fallback meaning.

Then I have updated a bit your action.php, (I will do a PR) to replace `og:image` by `og:image:secure_url`. I have chosen to "replace" and not to "add" because the accessed page is https or not. I mean we may have parts of the page in http, while asking https, but it's due to a problem in sending the response as everything should(must) be in https if asked so.
Moreover, having both urls means to do a trick in the `.htacess` to exclude the http file to be redirected to the https one...

At the end, it works with Twitter but not with facebook... It finds the meta, but don't display the right image... It seems that there are bugs (quite old) about this and I need to go deeper...
